jax/_src/scipy/special.py

```diff
@@ -1185,7 +1185,7 @@ def _sph_harm(m: Array,
   cos_colatitude = jnp.cos(phi)
 
   legendre = _gen_associated_legendre(n_max, cos_colatitude, True)
-  legendre_val = legendre.at[abs(m), n, jnp.arange(len(n))].get(mode="clip")
+  legendre_val = legendre.at[abs(m), n, jnp.arange(len(phi))].get(mode="clip")
```

Comment:
Yes it will break. But I observed that `scipy.special.shp_harm` and `jax.scipy.special.sph_harm` also raise [error](https://colab.sandbox.google.com/gist/rajasekharporeddy/816824b29f518e14e6ebdc79094f6037/-sph_harm.ipynb) when `n`, `phi` and `theta` are not broadcast compatible.

Reply:
Could we add code at the top of the function to explicitly check for whatever shape compatibility this function expects?

Reply:
Can we add `jax.lax.broadcast_shapes` at the top of the function to verify if the shapes of `m`, `n`, `theta`, and `phi` are broadcast compatible?

Reply:
If this is correct semantically, then yes this is how we should do it. Look at the uses of e.g. `jax._src.numpy.util.promote_args` elsewhere in the package.